Bubbkesort_week6/lib/bubble_sort.rb

```diff
@@ -0,0 +1,41 @@
+class BubbleSort
+  class InputOutputError < StandardError; end
+  attr_accessor :arr
+
+  def initialize(arr)
+    @arr = arr
+  end
+
+  def call
+    input_validation
+    sort
+  rescue InputOutputError => e
+    e.message
+  end
+
+  private
+
+  def sort
+    (0...arr.length).each do
+      (1..(arr.length - 1)).each do |idx|
```

Comment:
very deep. move inside loop to a new method

Reply:
fixed it